### PR TITLE
fix(db-engine-spec): execute oracle DML statement bug in sqllab

### PR DESCRIPTION
**Pull request by chuancyzhang**
when i execute oracle sql statements like update in SQLLAB, get "oracle error: not a query" error. 

Refer https://www.python.org/dev/peps/pep-0249/, superset old version use
`cursor.description` ，because this attribute will be None for operations that do not return rows or if the cursor has not had an operation invoked via the .execute*() method yet.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/oracle.py [hunk 1]
```diff
@@ -58,3 +63,18 @@ def epoch_to_dttm(cls) -> str:
...
+        :param limit: Maximum number of rows to be returned by the cursor
+        :return: Result of query
+        """
+        if not cursor.description:
+            return []
+        if cls.arraysize:
+            cursor.arraysize = cls.arraysize
+        if cls.limit_method == LimitMethod.FETCH_MANY:
+            return cursor.fetchmany(limit)
+        return cursor.fetchall()
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:80]: To avoid deuplicating code, do you think this work?
```suggestion
        super().fetch_data(cursor, limit)
```
Also, extra points for adding a unit test! 😉

**Review by villebro — CHANGES_REQUESTED** [review]
Small comment, other than that LGTM!

File: superset/db_engine_specs/oracle.py [hunk 2]
```diff
@@ -58,3 +63,14 @@ def epoch_to_dttm(cls) -> str:
     @classmethod
     def epoch_ms_to_dttm(cls) -> str:
         return "TO_DATE('1970-01-01','YYYY-MM-DD')+(1/24/60/60/1000)*{col}"
+    
+    @classmethod
+    def fetch_data(cls, cursor: Any, limit: int) -> List[Tuple]:
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:68]: The signature of this method has changed as of #10707 making `limit: Optional[int] = None`

**Review by villebro — CHANGES_REQUESTED** [review]

File: superset/db_engine_specs/oracle.py (hunk 2 above)
Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: OK，thank you very much.

Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: @villebro please help me check it.what is problem whit pre-commit?

File: superset/db_engine_specs/oracle.py [hunk 3]
```diff
@@ -58,3 +63,16 @@ def epoch_to_dttm(cls) -> str:
...
+    def fetch_data(
+        cls, cursor: Any, limit: Optional[int] = None
+    ) -> List[Tuple[Any, ...]]:
+        """
+        :param cursor: Cursor instance
+        :param limit: Maximum number of rows to be returned by the cursor
+        :return: Result of query
+        """
+        if not cursor.description:
+            return [()]
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:77]: This should be enough
```suggestion
            return []
```

File: superset/db_engine_specs/oracle.py [hunk 4]
```diff
@@ -58,3 +63,16 @@ def epoch_to_dttm(cls) -> str:
...
+        super().fetch_data(cursor, limit)
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:78]:
```suggestion
        return super().fetch_data(cursor, limit)
```

Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: Thanks

File: superset/db_engine_specs/oracle.py [hunk 5]
```diff
@@ -58,3 +63,16 @@ def epoch_to_dttm(cls) -> str:
     @classmethod
     def epoch_ms_to_dttm(cls) -> str:
         return "TO_DATE('1970-01-01','YYYY-MM-DD')+(1/24/60/60/1000)*{col}"
+
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:66]: Please check the linter's error message. There appears to be unnecessary whitespace on this row.

Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: OK，Thank you

Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: @villebro ,There seems to be a problem afflicting the pre-commit.

File: superset/db_engine_specs/oracle.py [hunk 6]
```diff
@@ -15,7 +15,12 @@
 # specific language governing permissions and limitations
 # under the License.
 from datetime import datetime
-from typing import Optional
+from typing import (
+    Any,
+    List,
+    Optional,
+    Tuple
+)
```
Comment (villebro) [inline superset/db_engine_specs/oracle.py:23]: This is what `isort` would do:
```suggestion
from typing import Any, List, Optional, Tuple
```
Check https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md for details on how to install pre-commit hooks to do this automatically:

```bash
pip3 install -r requirements/integration.txt
pre-commit install
```

Comment (chuancyzhang) [reply superset/db_engine_specs/oracle.py]: ok

**Review by villebro — APPROVED** [review]
LGTM, thanks for the fix and your patience!
